indent/vue.vim

```diff
@@ -6,59 +6,104 @@
-      \   { 'name': 'css', 'pairs': ['<style', '</style>'] },
-      \   { 'name': 'coffee', 'pairs': ['<script lang="coffee"', '</script>'] },
-      \   { 'name': 'javascript', 'pairs': ['<script', '</script>'] },
-      \ ]
+" HTML is included for <template> to apply policies set by
+" g:vue_indent_open_close_tags and g:vue_indent_first_line. Anything
+" outside pairs will also be handled by html.vim
+if !exists('g:vue_indent_languages')
+  let g:vue_indent_languages = [
+        \   {'name': 'pug', 'pairs': ['<template lang="pug"', '</template>']},
+        \   {'name': 'html', 'pairs': ['<template', '</template>'], 'first_line': 'html'},
+        \   {'name': 'stylus', 'pairs': ['<style lang="stylus"', '</style>']},
+        \   {'name': 'css', 'pairs': ['<style', '</style>']},
+        \   {'name': 'coffee', 'pairs': ['<script lang="coffee"', '</script>']},
+        \   {'name': 'javascript', 'pairs': ['<script', '</script>']},
+        \ ]
+endif
 
-for s:language in s:languages
-  " Set 'indentexpr' if the user has an indent file installed for the language
-  if strlen(globpath(&rtp, 'indent/'. s:language.name .'.vim'))
-    let s:language.indentexpr = s:get_indentexpr(s:language.name)
+if !exists('g:vue_indent_open_close_tags')
+  " Default of 0 forces open/close tags to the first column. Another
+  " reasonable option is 'html' which defers to html.vim to set the
+  " indentation.
+  let g:vue_indent_open_close_tags = 0
+endif
+
+if !exists('g:vue_indent_first_line')
+  " Default of -1 defers to autoindent which maintains the same indent as
+  " the previous line (the opening tag). During re-indentation, this also
+  " maintains whatever indent the user sets for the first line following
+  " the opening tag. Other reasonable options are 0 (force first column) or
+  " 'html' (defer to html.vim)
+  let g:vue_indent_first_line = -1
+endif
+
+" Load and return the indentexpr for language, resetting indentexpr to its
+" prior value before returning.
+function! s:get_indentexpr(language)
+  let saved_indentexpr = &indentexpr
+  " Default to blank to indicate that this language did not set indentexpr.
+  let &l:indentexpr = ''
+  if strlen(globpath(&rtp, 'indent/'. a:language .'.vim'))
+    unlet! b:did_indent
+    execute 'runtime! indent/' . a:language . '.vim'
+    let b:did_indent = 1
   endif
-endfor
+  let lang_indentexpr = &indentexpr
+  let &l:indentexpr = saved_indentexpr
+  return lang_indentexpr
+endfunction
 
 let s:html_indent = s:get_indentexpr('html')
 
-let b:did_indent = 1
-
-setlocal indentexpr=GetVueIndent()
+setlocal indentexpr=GetVueIndent(v:lnum)
 
 if exists('*GetVueIndent')
   finish
 endif
 
-function! GetVueIndent()
-  for language in s:languages
-    let opening_tag_line = searchpair(language.pairs[0], '', language.pairs[1], 'bWr')
+function! GetVueIndent(...)
+  let lnum = a:0 ? a:1 : getpos('.')[1]
+  let indent = s:get_vue_indent(lnum)
+  " NB: Strings compare as equal to 0, so 'html' == 0 is true, whoops.
```

Comment:
You could use `is#` here, which also checks if they are the same instance/type.

I think I will eventually `s/==/is/g` the entire codebase.